pennylane/devices/default_tensor.py

```diff
@@ -156,15 +160,43 @@ class DefaultTensor(Device):
             statistics like expectation values and variances is performed analytically.
         method (str): Supported method. Currently, only ``"mps"`` is supported.
         dtype (type): Datatype for the tensor representation. Must be one of ``np.complex64`` or ``np.complex128``.
```

Comment:
Does the dtype need to be provided in advance, or can it be inferred from user input?

Reply:
Excellent question, thanks for asking. This provided type is used for:

1) Setting the default value for `cutoff`
2) Setting the data type for the tensor in the `MPS_computational_state` function in `quimb`
3) Converting the type of matrices (each one corresponding to a specific operation) when applying gates

In particular, 3) is necessary since some operations in PennyLane have a matrix representation with an `int` type. If we try to provide this to `quimb` when applying gates to a quantum circuit, we get some cryptic errors (whose origin took time to be understood) such as:
```
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
```
Given the 3 points above, it seems to me that a data type parameter is the most transparent strategy to adopt. What do you think?

Reply:
Thanks @PietropaoloFrisoni! Sounds good, and we anyway have a default so it shouldn't cause much confusion with users.